salt/modules/junos.py

```diff
@@ -150,75 +150,50 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
-                conn.timeout = restore_timeout
-                raise
-        else:
-            return function(*args, **kwargs)
-
-    return wrapper
-
-
 def _timeout_decorator_cleankwargs(function):
+    """
+    Removing __pub kwargs during run and adding it back after to ensure non exposure.
+    Setting the timeout for the connection during run and then resetting the timeout.
+    Timeout has to be re-added for proxys running on the actual juniper
+    """
+
     @wraps(function)
     def wrapper(*args, **kwargs):
+        timeout_changed = False
+        old_kwargs = kwargs
+        # Loop through the kwargs and remove all __pub
+        del_kwargs = []
+        for key in kwargs:
+            if key.startswith("__pub"):
+                del_kwargs.append(key)
+        for del_kwarg in del_kwargs:
+            kwargs.pop(del_kwarg)
+
+        # Handle the timeout
         if "dev_timeout" in kwargs or "timeout" in kwargs:
-            ldev_timeout = max(kwargs.pop("dev_timeout", 0), kwargs.pop("timeout", 0))
             conn = __proxy__["junos.conn"]()
-            restore_timeout = conn.timeout
+            old_timeout = conn.timeout
+            # Default timeout is 60
+            # the int(x or 60) solves the problem where None is provided
+            ldev_timeout = max(
+                int(kwargs.pop("dev_timeout", 0) or 60),
+                int(kwargs.pop("timeout", 0) or 60),
+            )
             conn.timeout = ldev_timeout
-            try:
-                restore_kwargs = False
-                del_list = []
-                op = {}
-                op.update(kwargs)
-                for keychk in kwargs:
-                    if keychk.startswith("__pub"):
-                        del_list.append(keychk)
-                if del_list:
-                    restore_kwargs = True
-                    for delkey in del_list:
-                        kwargs.pop(delkey)
-
-                result = function(*args, **kwargs)
-                if restore_kwargs:
-                    kwargs.update(op)
-
-                conn.timeout = restore_timeout
-                return result
-            except Exception:  # pylint: disable=broad-except
-                conn.timeout = restore_timeout
-                raise
-        else:
-            restore_kwargs = False
-            del_list = []
-            op = {}
-            op.update(kwargs)
-            for keychk in kwargs:
-                if keychk.startswith("__pub"):
-                    del_list.append(keychk)
-            if del_list:
-                restore_kwargs = True
-                for delkey in del_list:
-                    kwargs.pop(delkey)
-
-            ret = function(*args, **kwargs)
-            if restore_kwargs:
-                kwargs.update(op)
+            timeout_changed = True
 
-            return ret
+        # Run the function
+        try:
+            result = function(*args, **kwargs)
+        except:
+            raise
```

Comment:
```suggestion
```
Don't catch exceptions just to raise them afterwards without doing anything with that information.

Reply:
It was done like that before therefor I keept the same functionality.

Reply:
I have to disagree.
Neither [this](https://github.com/saltstack/salt/blob/3006.x/salt/modules/junos.py#L153-L169) chunk, nor [this](https://github.com/saltstack/salt/blob/3006.x/salt/modules/junos.py#L174-L203) has an exception caught just to be raised immediately.
There's code that runs, before raising the exception again.

Please remove that bare except, you're not doing anything with it.